src/webots/gui/WbDragSolidEvent.cpp

```diff
@@ -235,26 +234,19 @@ void WbDragForceEvent::updateOrigin() {
 }
 
 void WbDragForceEvent::applyToOde() {
-  mScalingFactor = FORCE_SCALING_FACTOR * mVector.length2();
   // ODE
   mSelectedSolid->awake();
-  mSelectedSolid->addForceAtPosition(mScalingFactor * mVector, mOrigin);
+  mSelectedSolid->addForceAtPosition(mScalingFactor * mVector.length2() * mVector, mOrigin);
 }
 
 QString WbDragForceEvent::magnitudeString() {
-  const double lengthSquared = mVector.length2();
-  mScalingFactor = FORCE_SCALING_FACTOR * lengthSquared;
-  mMagnitude = mScalingFactor * sqrt(lengthSquared);
-  QString returnValue = WbPrecision::doubleToString(mMagnitude, WbPrecision::GUI_MEDIUM) + " N";
-  mMagnitude = mScalingFactor * mVector.length();
-  return returnValue;
+  mMagnitude = mScalingFactor * mVector.length2() * mVector.length();
+  return WbPrecision::doubleToString(mMagnitude, WbPrecision::GUI_MEDIUM) + " N";
 }
```

Comment:
Code clean-up: I would delete this apparently useless `mMagnituteString` field and make this method const.
The same applies to the `WbDragTorqueEvent` object.